src/main/java/com/example/java/maven/ticketVendingMachine/CoinStorage.java

```diff
@@ -0,0 +1,86 @@
+package com.example.java.maven.ticketVendingMachine;
+
+import java.util.Map;
+import java.util.Optional;
+
+public class CoinStorage {
+    private final Map<Coin, Integer> coins;
+    private final MessagePrinter messagePrinter;
+
+    public CoinStorage(Map<Coin, Integer> coins, MessagePrinter messagePrinter) {
+        this.coins = coins;
+        this.messagePrinter = messagePrinter;
+    }
+
+    public Map<Coin, Integer> getCoins() {
+        return coins;
+    }
+
+    public void clear() {
+        coins.clear();
+    }
+
+    public void addCoin(Coin coin) {
+        int currentNumberOfCoins = Optional.ofNullable(coins.get(coin)).orElse(0);
```

Comment:
can be replaced with `getOrDefault()` but still, nice usage of optional!

Reply:
`getOrDefault()` to remember.